services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -101,6 +106,58 @@ func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*Googl
 	return &GooglePubSubProducer{client: &PubsubClient{client, topicMap, o}, conf: config.Default}, nil
 }
 
+func (producer *GooglePubSubProducer) publish(ctx context.Context, topic *pubsub.Topic, message *pubsub.Message) (string, error) {
+	result := topic.Publish(ctx, message)
+	serverID, err := result.Get(ctx)
+	if err != nil {
+		return "", err
+	}
+	return serverID, nil
+}
+
+// publishWithRetry publishes a message with retry logic for intermittent authentication errors
+func (producer *GooglePubSubProducer) publishWithRetry(ctx context.Context, topic *pubsub.Topic, message *pubsub.Message) (string, error) {
+	pkgLogger.Infof("[GooglePubSub] Publishing with retry enabled")
+	var serverID string
+
+	operation := func() error {
+		var err error
+		serverID, err = producer.publish(ctx, topic, message)
+		if err == nil {
+			return nil
+		}
+
+		// Only retry on authentication errors
+		st, ok := status.FromError(err)
+		if ok && st.Code() == codes.Unauthenticated {
+			pkgLogger.Warnf("[GooglePubSub] Authentication error, retrying: %v", err)
+			return err // Return error to trigger retry
+		}
+
+		// For non-authentication errors, mark as permanent to avoid retry
+		return backoff.Permanent(err)
+	}
+
+	// Use exponential backoff with exactly 3 attempts
+	bo := backoff.NewExponentialBackOff()
+	bo.InitialInterval = 100 * time.Millisecond
+	bo.MaxInterval = 1 * time.Second
+	bo.MaxElapsedTime = 1 * time.Second
+	bo.Multiplier = 2
```

Comment:
please make this configurable and use reloadable config variables in `GooglePubSubProducer`

```
Router.GOOGLEPUBSUB.Client.Retry.InitialInterval: 10ms
Router.GOOGLEPUBSUB.Client.Retry.MaxInterval: 500ms
Router.GOOGLEPUBSUB.Client.Retry.MaxElapsedTime: 2s
Router.GOOGLEPUBSUB.Client.Retry.MaxRetries: 3
```